### DISCUSSION TITLE
EdgeX Tooling Poll

### DISCUSSION THREAD

**Discussion by JamesKButcher** [edgexfoundry/community#34] [Polls]
Hi EdgeX Community!

I’m looking for some responses from the users of EdgeX as to how they interact with EdgeX please…

Obviously we have well defined APIs for controlling and interfacing with EdgeX. But we also have a Command Line Interface (CLI) and a Graphical User Interface (GUI). I’d like to hear whether people generally use all of the above or whether they focus on using just the APIs and/or create their own tooling. This info will help with our focus going forward.

Looks like we can only select one answer with the poll tool here, hence all my different options below! But please leave any additional info in the comments.

Thanks all,
James Butcher
EdgeX TSC Chair

**Comment by ed-lisle** [comment]
Plus our own tooling - so all of the above.

**Comment by sounak-1989** [comment]
We use Postman to access all the APIs. We usually run EdgeX on the target machine or the development machine(WSL) and open the ports for access within the network. Then we access the services through API calls using Postman.

**Comment by girigupt** [comment]
Each method of access is required since each of them cater to different audience: API ( postman collection) is good for developers, CLI is for devOps/QA team while UI is for end users/customers. I know current UI is not meant for production use, so we have developed our own UI that provides a consolidated view of profile/devices across multiple deployments ( data is synced)

**Comment by mohammedimrankasab** [comment]
@JamesKButcher Can you please guide me where can I find the CLI tool for EdgeX?

**Reply by JamesKButcher** [reply]
Hi @mohammedimrankasab 

Following the information obtained above, and with no forthcoming community resources to help bring it forward, the EdgeX TSC took the decision to archive the EdgeX CLI and not bring it forward to EdgeX V3.

The subject was also raised here: https://github.com/orgs/edgexfoundry/discussions/61 and confirmed via the EdgeX-TSC mailing list, but for completeness I will add another comment on that discussion too

Thanks
James

**Reply by mohammedimrankasab** [reply]
Thank you @JamesKButcher for your quick reply, it is really helpful 👍🏼
